types/index.d.ts

```diff
@@ -143,17 +143,44 @@ declare namespace Realm {
     }
 
     enum ClientResetMode {
-        Manual = "manual",
-        DiscardLocal = "discardLocal",
+        Manual = 'manual',
+        DiscardLocal = 'discardLocal',
+        Recover = 'recover',
+        RecoverOrDiscard = 'recoverOrDiscard'
     }
 
+    enum ClientResetDidRecover {
+      Yes = 'yes',
+      No = 'no'
+    }
```

Comment:
Why is this not just a simple `boolean`?

Reply:
It would allow for latter adding "perhaps - not entirely sure" 🤷

Reply:
`undefined` would be a great way to express that in that theoretical future.

I suggest making this a boolean for now if we're not certain that new values will be added soon.

Alternatively I'd suggest using values that are closer to semantics:

```suggestion
    enum ClientResetOutcome {
      Recovered = 'recovered',
      Discarded = 'discarded'
    }
```

Reply:
Another alternative (implemented in other SDKs) is to provide a third callback (similar to the callback for manual mode) which is called when recovery is not possible - or when an error occurs.